ipalib/frontend.py

```diff
@@ -1298,7 +1298,10 @@ def __json__(self):
         )
         if self.primary_key:
             json_dict['primary_key'] = self.primary_key.name
-        json_dict['methods'] = list(self.methods)
+        json_dict['methods'] = [
+            name for name, method in self.methods.__todict__().items()
+            if not isinstance(method, Local)
```

Comment:
Interestingly ipaserver/plugins/internal.py::json_metadata is already trying to filter these out.
I suppose if this is done at a lower level then the instance checking can be removed from json_metadata().

Reply:
I think we still need it in higher levels when processing Method objects. This just filters out method names.